File: src/com/quotegraph/view/UserInterface.java
package com.quotegraph.view;

import java.awt.BorderLayout;
import javax.swing.JFrame;
import javax.swing.WindowConstants;

/**
 * Create a JFrame container window and add a dropdown.
 *
 * @author Daniel Peters
 * @version 1.0
 */
public class UserInterface extends JFrame {
  /**
   * Reference to the data select dropdown.
   */
  private final DataSelect dropdown;

  /**
   * Initializes window look and dropdown.
   *
   * @param dropdown Reference to the select dropdown object
   */
  public UserInterface(DataSelect dropdown) {
    this.dropdown = dropdown;
    initAppearance();
  }

  /**
   * Initialize the looks and title of the window.
   */
  private void initAppearance() {
    this.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
    this.setLayout(new BorderLayout());
    this.setTitle("Graph Panel");
    this.add(dropdown, BorderLayout.NORTH);
    this.setLocationByPlatform(true);
  }
}

File: src/main/kotlin/ch/peters/daniel/quotegraph/view/UserInterface.kt
package ch.peters.daniel.quotegraph.view

import java.awt.BorderLayout
import javax.swing.JFrame
import javax.swing.WindowConstants

/**
 * Create a JFrame container window and add a dropdown.
 *
 * @author Daniel Peters
 * @version 2.0
 */
class UserInterface(dropdown: DataSelect) : JFrame() {
  init {
    this.defaultCloseOperation = WindowConstants.EXIT_ON_CLOSE
    this.layout = BorderLayout()
    this.title = "Graph Panel"
    this.add(dropdown, BorderLayout.NORTH)
    this.isLocationByPlatform = true
  }
}
